### DISCUSSION TITLE
revert to 'classics' shortcuts?

### DISCUSSION THREAD

**Discussion by ecooch** [WinEdt-Team/WinEdt-Support#45] [1.1 General Help and Discussion]
I just did an in-place update to the latest build of 11.2. However, in so doing, the 'update' blew away to key macros/shortcuts that I have been using (for years, and years), and replaced them with 'new defaults'. 

For example ctrl-shift-L now start LuaLatex, and not generic Latex (which it used to).  ctrl-shift-S used to take me from the source file to the corresponding point in the DVI. Now, ctrl-shift-S does....nothing at all.

So, how do I get all the old ctrl-shift key shortcuts back to the 'original' (classic) settings?

**Comment by urmel1960** [comment]
Have you tried running "Options/Maintenance/Upgrade Local" ?

**Reply by ecooch** [reply]
Yes -- I did that as part of the usual upgrade sequence.

**Comment by Alex-Simonic** [comment]
You were probably using the default menu without any custom modifications. The old default settings have been replaced with the new ones when you executed Upgrade Local. Users who modified the menu would be given a choice to preserve their customized menus or replace them with the new default ones (as they wish).

The new default TeX menu favors PDF/Unicode TeX engines and a few shortcuts have been reassigned accordingly. There is nothing wrong with this...

It is easy to customize your shortcuts to what you are accustomed. Start the Options Interface and double-click on the Main Menu item. Search for:

ITEM="LuaLaTeX"

There you will find:

   ` SHORTCUT="24652::Ctrl+Shift+L"`

Comment it out or else double-click on it and replace it with something else (like Ctrl+L). Now scroll down to plain LaTeX (ITEM="LaTeX"), double-click on the SHORTCUT, and enter Ctrl+Shift+L. Reload the modified menu by clicking on the first item in the Options Interface's Toolbar (Load Current Script).

You can do the same with any other `Ctrl+Shift` shortcut you want to change. There is one more thing: when you enter a shortcut you should do the Search to see that it is not already used before this one in the Main Menu (the first one prevails if there is a conflict). For example, `Ctrl+Shift+S` is used for Save All and you will have to comment it out before you can use it for Forward DVI Search.

Now you know how you can modify this. Changing a handful (3 or 4) of shortcuts in the TeX Menu should only take a few minutes. Your modifications will be preserved for future upgrades. The so-called "classic" shortcuts were assigned decades ago when LaTeX, TeX, and AMSTeX were the only game in town. The new TeX Menu (with a few shortcut changes) was introduced with new users in mind but you can restore your shortcuts if you follow the instructions above.

**Comment by ecooch** [comment]
> You were probably using the default menu without any custom modifications. The old default settings have been replaced with the new ones when you executed Upgrade Local. Users who modified the menu would be given a choice to preserve their customized menus or replace them with the new default ones (as they wish).

When I did the upgrade, 'Upgrade Local' did *not* prompt me to save or replace current settings. Which I think caused the problem -- it defaulted to overwriting my shortcut settings (and various other things I suspect...). 

While you provided the mechanics I'd need to run through to manually change the shortcuts (thanks!) it would be much simpler if there was a single file I could copy from another machine -- which has the old shortcuts I'm used to -- onto the machine that I just upgraded.  Is there such a file? I maintain multiple duplicated installs of software (including WinEdt) for just this sort of contingency. 

Thanks in advance...

**Reply by ecooch** [reply]
I'm assuming the 'file' in question is MainMenu.ini. I'll simply copy the file from one of my other systems (that had the full set of 'old' shortcuts) onto the system where I did the 'upgrade', and see how that works.

**Reply by Alex-Simonic** [reply]
You were not prompted because you were using the default menu settings: you had no custom shortcut settings.

If you follow the instructions above you will end up with a custom MainMenu.ini. This is the file you have to copy. If you start copying the default .ini files from the install folder from an older build you will have to manually take care of interdependencies (like Toolbar). Files in the install folder are not preserved during upgrades (that is on purpose). If you don't want defaults you should have a local instance of MainMenu.ini with shortcuts of your choice. If you are using the defaults and the default settings change you will inherit the changes (whether you want it or not). I suppose you can copy old MainMeni.ini in your local settings and execute Upgrade Local but there is no guarantee that this will work for you (or that you will copy the right file to the right place). I have nothing to add to this but if somebody else has the patience to guide you further that would be fine with me...
Fair enough?

**Reply by urmel1960** [reply]
If you would like to have all your settings in a single file in future, please take a look at the "Private Settings" section in "Advanced Configuration" in Winedt Help. It describes exactly how you can do this and what you need to bear in mind. I recently did this for my installation as well and am very happy to have all the customizations in one place.

Depending on which version you upgraded from (you wrote something about functionalities that you have been using for years), I would carry out the adjustments again that you are now missing instead of copying old ini files and possibly having to work through side effects.

**Comment by ecooch** [comment]
Well, now I'm really confused. For convenience, I'll refer to the two computers as 'work' and 'home'. This morning, on my 'work' machine, I updated to the latest build of WinEdt -- which precipitated the problems I had. My 'home' machine (which I didn't update) still 'works' with old-school shortcuts. [Both are WinEdt 11.2]. 

So, as per above, thought it worth seeing what differences there might be in MainMenu.ini.  Much to my surprise, there are *no* differences at all between the MainMenu.ini on either my 'work' or 'home' machines.  If I look at the MainMenu.ini on my 'home' machine, I see the following for (say) 'Latex':

         ITEM="LaTeX"
         CAPTION="&LaTeX"
         IMAGE="TeXLaTeX"
         SAVE_INPUT=1
         MACRO="Exe('%b\Exec\TeX\LaTeX.edt');"
         SHORTCUT="16460::Ctrl+L"

So, the *new* shortcut (ctrl-L), even though

1\ ctrl-shift-L  works!!

and (more puzzling)

2\  if I hover over the 'L' button in the toolbar (which I would click if I wanted a straight LaTeX compilation, it shows (below) that it is associated with ctrl-shift-L. 

![Untitled-1](https://github.com/WinEdt-Team/WinEdt-Support/assets/4500611/6b3c8957-56d1-42c2-8eeb-c94f7d39d7bf)

So, somewhere in the WinEdt install on my home machine, **all** of the 'old' shortcuts are specified, and hooked to the menu buttons! For example, if I hover over the DVI search button, it shows ctrl-shift-S.  But, MainMenu.ini shows a completely different shortcut:

      ITEM="DVI_Search"
      CAPTION="DVI &Search"
      IMAGE="DviFind"
      MACRO="Exe('%b\Exec\TeX\DVI Search.edt');"
      SHORTCUT="41079::Alt+Shift+F8"    

So, where the heck is WinEdt pulling the shortcuts from on my 'home' machine? There must be some .ini file somewhere that is over-riding the MainMenu.ini file. but....where?

**Reply by Alex-Simonic** [reply]
They are stored nowhere. On your home machine, you did not rebuild local settings so your old shortcuts are still loaded (in local WinEdt.dnt) since the new default MainMenu.ini was never loaded AND you do not have a (customized) local one. You don't seem to understand much about the default and local settings. The advice I gave you is the best way to get your "problem" solved. Changing 4 shortcuts would be much faster than all this correspondence... I could (but will not) make a download that would restore old shortcuts into the new menu since you can do it yourself in less than a minute and I don't think it is a big issue for most other users.

**Reply by Alex-Simonic** [reply]
Just in case you (or anyone else) want to restore the old WinEdt 11.1 menu (and shortcuts) here it is:

[MainMenu11.1.zip](https://github.com/WinEdt-Team/WinEdt-Support/files/15369886/MainMenu11.1.zip)

Unzip MainMenu.ini in your LOCAL ConfigEx folder and execute Upgrade Local. Now you have your "classic" shortcuts exactly as they were. Case closed.

**Comment by ecooch** [comment]
Perfect. Many thanks for taking the time to pull a 'solution' together.

**Comment by ecooch** [comment]
So, I just installed latest WinEdt on a new machine.  Configured it for '2 -row small'. All buttons as they should be. But, when  put MainMenu.ini into LOCAL COnfigEx and Upgrade Local (as per directions), the 'dvi2ps' button disappears completely, leaving a gap in the toolbar:

![missing_button](https://github.com/user-attachments/assets/25f0c5ce-19d6-4513-91b8-f6503bff84f0)

This is repeatable. When I uninstall WinWEdt, re-install from scratch, 2-row toolbar, button is there. But the moment I rebuild local, this one button gets clobbered.

**Comment by ecooch** [comment]
Solved. I was in fact using the wrong MainMenu.ini file. My mistake.
